citesphere/src/main/java/edu/asu/diging/citesphere/api/v1/apps/ApplicationsInfoController.java

```diff
@@ -0,0 +1,106 @@
+package edu.asu.diging.citesphere.api.v1.apps;
+
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import edu.asu.diging.citesphere.api.v1.V1Controller;
+import edu.asu.diging.citesphere.core.model.Role;
+import edu.asu.diging.citesphere.core.model.oauth.impl.OAuthClient;
+import edu.asu.diging.citesphere.core.service.oauth.IOAuthClientManager;
+
+@Controller
+public class ApplicationsInfoController extends V1Controller {
+
+    @Autowired
+    private IOAuthClientManager clientManager;
+    
+    @GetMapping("/admin/apps")
+    public ResponseEntity<List<AppInfo>> getAllApps(Principal principal) {
+
+        //Checks if the user is of role trusted client
+        boolean flag = false;
+        if (principal instanceof OAuth2Authentication) {
+            OAuth2Authentication userDetails = (OAuth2Authentication) principal;
+            for (GrantedAuthority authority : userDetails.getAuthorities()) {
+                if (authority.getAuthority().equals(Role.TRUSTED_CLIENT)) {
+                    flag = true;
+                    break;
+                }
+            }
+        }
+
+        //If the user is not of expected role, respond as unauthorized access 
+        if (!flag)
+            return new ResponseEntity<List<AppInfo>>(HttpStatus.UNAUTHORIZED);
```

Comment:
needs to follow our coding style.

Reply:
I'm not sure which coding style is being violated here.

Reply:
please read our code style